### ISSUE TITLE
Multitenant application and replica connection database name switching with "USE database_name;"

### ISSUE THREAD

**Issue by lizdeika** [rails/rails#37762]
### Steps to reproduce
In a multitenant application we use one database server with multiple databases inside for primary connection. And another server with multiple databases for replica connections.
Database name is choosen by the domain.
We then do `ActiveRecord::Base.connection.execute('USE database_name_for_domain')`.
The same is needed for replica connections, but: 

I am not able to do this on replica connection:
```ruby
ActiveRecord::Base.connected_to(role: :read1) do
  ActiveRecord::Base.connection.execute("USE database_name_2")
end
```
Because replica connections are preventing writes in `active_record/connection_adapters/mysql/database_statements.rb`
```ruby
def execute(sql, name = nil)
  if preventing_writes? && write_query?(sql)
    raise ActiveRecord::ReadOnlyError, "Write query attempted while in readonly mode: #{sql}"
  end
  ...
```

To workaround this I made a monkey patch aliasing an unrestricted execute method from `AbstractMysqlAdapter` for later use:
```ruby
require 'active_record/connection_adapters/abstract_mysql_adapter'

module ActiveRecord
  module ConnectionAdapters
    class AbstractMysqlAdapter < AbstractAdapter
      alias unrestricted_execute :execute
    end
  end
end
```

So I am able to do this:
```ruby
ActiveRecord::Base.connected_to(role: :read1) do
  ActiveRecord::Base.connection.unrestricted_execute("USE database_name_2")
end
```

Is there a non-monkey-patch way to achieve this?
I am not sure `USE` should be restricted to execute for replica connections.

### System configuration
**Rails version**: 6.0.1

**Ruby version**: 2.5.1

**Database**: MySQL

**Comment by eileencodes** [comment]
We could add `use` to the regex list for allowed read queries, but I don't know if that's a normal use case. In my own local app I use for testing multiple databases my replica users grants are set to select. In my case even when I add `use` it won't work because the replica user is not allowed to run `use`. What grants do your readonly users have?

This seems like a valid use case but I want more information before adding tons of stuff to the regex.

**Comment by lizdeika** [comment]
Hi,
We have only `select` granted for replica users.
I think that when user is granted `select` privilege to some database he can also do the `USE` statement. Because one can do, when granted `select` to both `db1` and `db2`:
```sql
USE db1;
SELECT author_name,editor_name FROM author,db2.editor
WHERE author.editor_id = db2.editor.editor_id;
```
(example from https://dev.mysql.com/doc/refman/5.7/en/use.html)

**Comment by eileencodes** [comment]
Fixed by #38042
